src/ota/mod.rs

```diff
@@ -104,3 +126,2108 @@ impl Default for DeployStatus {
+
+        match signal {
+            0 => {
+                info!("Update successful");
+
+                OtaStatus::Deployed(ota_request)
+            }
+            _ => {
+                let message = format!("Update failed with signal {signal}",);
+
+                match self.last_error().await {
+                    Ok(err) => {
+                        error!("{message}: {err}");
+                    }
+                    Err(err) => {
+                        error!("{message}: {}", stable_eyre::Report::new(err));
+                    }
+                }
+
+                OtaStatus::Failure(OtaError::InvalidBaseImage(message), Some(ota_request))
+            }
+        }
+    }
+
+    /// Handle the transition to rebooting status.
+    pub async fn reboot(&self, ota_request: OtaId) -> OtaStatus {
+        if self
+            .publisher_tx
+            .send(OtaStatus::Rebooting(ota_request.clone()))
+            .await
+            .is_err()
+        {
+            warn!("ota_status_publisher dropped before sending rebooting_status")
+        };
+
+        info!("Rebooting the device");
+
+        #[cfg(not(test))]
+        if let Err(error) = crate::power_management::reboot().await {
+            let message = "Unable to run reboot command";
+            error!("{message} : {error}");
+            return OtaStatus::Failure(OtaError::Internal(message), Some(ota_request.clone()));
+        }
+
+        OtaStatus::Rebooted
+    }
+
+    /// Handle the transition to success status.
+    pub async fn check_reboot(&self) -> OtaStatus {
+        if !self.state_repository.exists().await {
+            return OtaStatus::NoPendingOta;
+        }
+
+        info!("Found pending update");
+
+        let ota_state = match self.state_repository.read().await {
+            Ok(state) => state,
+            Err(err) => {
+                let message = "Unable to read pending ota state".to_string();
+                error!("{message} : {}", err);
+                return OtaStatus::Failure(OtaError::Io(message), None);
+            }
+        };
+
+        let request_uuid = ota_state.uuid;
+        let ota_request = OtaId {
+            uuid: request_uuid,
+            url: "".to_string(),
+        };
+
+        if let Err(error) = self.do_pending_ota(&ota_state).await {
+            return OtaStatus::Failure(error, Some(ota_request));
+        }
+
+        OtaStatus::Success(ota_request)
+    }
+
+    pub async fn do_pending_ota(&self, state: &PersistentState) -> Result<(), OtaError> {
+        const GOOD_STATE: &str = "good";
+
+        let booted_slot = self.system_update.boot_slot().await.map_err(|error| {
+            let message = "Unable to identify the booted slot";
+            error!("{message}: {error}");
+            OtaError::Internal(message)
+        })?;
+
+        if state.slot == booted_slot {
+            let message = "Unable to switch slot";
+            return Err(OtaError::SystemRollback(message));
```

Comment:
Can't you just inline the message here since you are not logging?

Reply:
Added log